code/modules/mining/minebot.dm

```diff
@@ -1,313 +1,699 @@
+#define MODE_MINING "mining" /// Ore collection/mining mode for AI, plasma cutter usage mode for players
 
 /mob/living/simple_animal/hostile/mining_drone
-	name = "\improper Nanotrasen minebot"
-	desc = "The instructions printed on the side read: This is a small robot used to support miners, can be set to search and collect loose ore, or to help fend off wildlife."
+	name = "minebot"
+	desc = "A small robot used to support miners. It can be set to search and collect loose ore, mine any ore it detects, or help fend off wildlife. It is equipped with a mining drill and kinetic accelerator, with mounting points for a plasma cutter."
 	gender = NEUTER
 	icon = 'icons/mob/aibots.dmi'
 	icon_state = "mining_drone"
 	icon_living = "mining_drone"
+	icon_dead = "mining_drone_disabled"
 	status_flags = CANSTUN|CANKNOCKDOWN|CANPUSH
 	mouse_opacity = MOUSE_OPACITY_ICON
 	faction = list("neutral")
 	a_intent = INTENT_HARM
+	hud_type = /datum/hud/minebot
+	// Atmos
 	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
 	minbodytemp = 0
-	move_to_delay = 10
+	// Health/damage
 	health = 125
 	maxHealth = 125
 	melee_damage = 15
 	obj_damage = 10
 	environment_smash = ENVIRONMENT_SMASH_NONE
+	healable = 0
+	deathmessage = "stops moving"
+	// AI stuff
 	check_friendly_fire = TRUE
-	stop_automated_movement_when_pulled = TRUE
-	attacktext = "drills"
-	attack_sound = 'sound/weapons/circsawhit.ogg'
+	move_to_delay = 10
+	ranged = TRUE
 	sentience_type = SENTIENCE_MINEBOT
-	speak_emote = list("states")
+	stop_automated_movement_when_pulled = TRUE
+	wander = FALSE
 	wanted_objects = list(/obj/item/stack/ore/diamond, /obj/item/stack/ore/gold, /obj/item/stack/ore/silver,
 						  /obj/item/stack/ore/plasma, /obj/item/stack/ore/uranium, /obj/item/stack/ore/iron,
 						  /obj/item/stack/ore/bananium, /obj/item/stack/ore/titanium)
-	healable = 0
-	loot = list(/obj/effect/decal/cleanable/robot_debris)
-	del_on_death = TRUE
+	// Response verbs
+	response_help = "pets"
+	attacktext = "drills"
+	attack_sound = 'sound/weapons/circsawhit.ogg'
+	speak_emote = list("states")
+	// Light handling
 	light_system = MOVABLE_LIGHT
 	light_range = 6
 	light_on = FALSE
-	var/mode = MINEDRONE_COLLECT
-	var/obj/item/gun/energy/kinetic_accelerator/minebot/stored_gun
+	// Minebot-specific vars
+	var/mode = MODE_MINING /// What mode the minebot is in
+	var/mining_enabled = FALSE /// Whether or not the minebot will mine new ores while in mining mode.
+	var/list/installed_upgrades /// A list of all the minebot's installed upgrades
+	var/obj/item/gun/energy/kinetic_accelerator/minebot/stored_pka /// The minebot's stored PKA
+	var/obj/item/gun/energy/plasmacutter/stored_cutter /// The minebot's stored plasma cutter
+	var/obj/item/pickaxe/drill/stored_drill /// The minebot's stored drill
+	var/obj/item/t_scanner/adv_mining_scanner/stored_scanner /// The minebot's stored mining scanner
```

Comment:
These strong references can result in harddels

Reply:
I added some signals to clear them if they get deleted, let me know if I missed anything.